missing-field-initializers in test/api/test-ot-face.c

serval2412 opened the pull request:
FAILED: test/fuzzing/hb-shape-fuzzer.p/hb-shape-fuzzer.cc.o clang++ -Itest/fuzzing/hb-shape-fuzzer.p -Itest/fuzzing -I../test/fuzzing -I. -I.. -Isrc -I../src -fdiagnostics-color=always -D_GLIBCXX_ASSERTIONS=1 -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c++11 -fno-exceptions -O3 -fno-exceptions -fno-rtti -fno-threadsafe-statics -fvisibility-inlines-hidden -DHAVE_CONFIG_H -Wall -Wextra -Werror -MD -MQ test/fuzzing/hb-shape-fuzzer.p/hb-shape-fuzzer.cc.o -MF test/fuzzing/hb-shape-fuzzer.p/hb-shape-fuzzer.cc.o.d -o test/fuzzing/hb-shape-fuzzer.p/hb-shape-fuzzer.cc.o -c ../test/fuzzing/hb-shape-fuzzer.cc In file included from ../test/fuzzing/hb-shape-fuzzer.cc:9: ../test/fuzzing/../api/test-ot-face.c:47:34: error: missing field 'y_bearing' initializer [-Werror,-Wmissing-field-initializers]
  hb_glyph_extents_t extents = {0};
                                 ^
../test/fuzzing/../api/test-ot-face.c:85:55: error: missing field 'enable' initializer [-Werror,-Wmissing-field-initializers]
    hb_aat_layout_feature_selector_info_t setting = {0};
                                                      ^
2 errors generated.

serval2412 commented:
Just for information, it's the last one I got with missing-field-initializers

behdad commented:
> Just for information, it's the last one I got with missing-field-initializers

Feel free to enable it as error in `hb.hh` then.

behdad commented:
Humm. I thought I merged this. What happened?

serval2412 commented:
I don't know what happened with the merge, I've already deleted the branch.
Anyway I removed locally #pragma GCC diagnostic ignored "-Wmissing-field-initializers" and I see there are a lot more.
I don't understand why clang didn't see them first.
I'll give it a new try but it'll require more time.

behdad commented:
Humm. Might not be worth it then, since technically these are well-defined and not an error.
